runtime/centrifuge/src/lib.rs

```diff
@@ -1000,6 +1121,425 @@ impl pallet_collator_selection::Config for Runtime {
+				),
+				_ => false,
+			}
+		} else {
+			false
+		}
+	}
+}
+
+impl pallet_permissions::Config for Runtime {
+	type AdminOrigin = EnsureRootOr<HalfOfCouncil>;
+	type Editors = Editors;
+	type MaxRolesPerScope = MaxRolesPerPool;
+	type Role = Role<TrancheId, Moment>;
+	type RuntimeEvent = RuntimeEvent;
+	type Scope = PermissionScope<PoolId, CurrencyId>;
+	type Storage =
+		PermissionRoles<TimeProvider<Timestamp>, MinDelay, TrancheId, MaxTranches, Moment>;
+	type WeightInfo = weights::pallet_permissions::WeightInfo<Runtime>;
+}
+
+parameter_types! {
+	pub const MaxOutstandingCollects: u32 = 10;
+}
+
+/// Checks whether the given `who` has the role
+/// of a `TrancehInvestor` for the given pool.
+pub struct IsTrancheInvestor<P, T>(PhantomData<(P, T)>);
+impl<
+		P: PermissionsT<AccountId, Scope = PermissionScope<PoolId, CurrencyId>, Role = Role>,
+		T: UnixTime,
+	> PreConditions<OrderType<AccountId, TrancheCurrency, Balance>> for IsTrancheInvestor<P, T>
+{
+	type Result = DispatchResult;
+
+	fn check(order: OrderType<AccountId, TrancheCurrency, Balance>) -> Self::Result {
+		let is_tranche_investor = match order {
+			OrderType::Investment {
+				who,
+				investment_id: tranche,
+				..
+			} => P::has(
+				PermissionScope::Pool(tranche.of_pool()),
+				who,
+				Role::PoolRole(PoolRole::TrancheInvestor(
+					tranche.of_tranche(),
+					T::now().as_secs(),
+				)),
+			),
+			OrderType::Redemption {
+				who,
+				investment_id: tranche,
+				..
+			} => P::has(
+				PermissionScope::Pool(tranche.of_pool()),
+				who,
+				Role::PoolRole(PoolRole::TrancheInvestor(
+					tranche.of_tranche(),
+					T::now().as_secs(),
+				)),
+			),
+		};
+
+		if is_tranche_investor {
+			Ok(())
+		} else {
+			// TODO: We should adapt the permissions pallets interface to return an error instead of a boolen. This makes the redundant has not role error
+			//       that downstream pallets always need to generate not needed anymore.
+			Err(DispatchError::Other(
+				"Account does not have the TrancheInvestor permission.",
+			))
+		}
+	}
+}
+
+impl pallet_investments::Config for Runtime {
+	type Accountant = PoolSystem;
+	type Amount = Balance;
+	type BalanceRatio = Rate;
+	type InvestmentId = TrancheCurrency;
+	type MaxOutstandingCollects = MaxOutstandingCollects;
+	type PreConditions = IsTrancheInvestor<Permissions, Timestamp>;
+	type RuntimeEvent = RuntimeEvent;
+	type Tokens = Tokens;
+	type WeightInfo = ();
+}
+
+impl pallet_interest_accrual::Config for Runtime {
+	type Balance = Balance;
+	type InterestRate = Rate;
+	// TODO: This is a stopgap value until we can calculate it correctly with updated benchmarks. See #1024
+	type MaxRateCount = MaxActiveLoansPerPool;
+	type RuntimeEvent = RuntimeEvent;
+	type Time = Timestamp;
+	type Weights = weights::pallet_interest_accrual::WeightInfo<Self>;
+}
+
+parameter_types! {
+	pub const LoansPalletId: PalletId = cfg_types::ids::LOANS_PALLET_ID;
+	pub const MaxActiveLoansPerPool: u32 = 1000;
```

Comment:
Are we ok with 1000 or we need a higher value here?

Reply:
Now that benches have been run on this PR we can probably update this before merge?